skyportal/handlers/api/gcn.py

```diff
@@ -0,0 +1,202 @@
+        """
+        ---
+        description: Ingest GCN xml file
+        responses:
+          200:
+            content:
+              application/json:
+                schema: Success
+          400:
+            content:
+              application/json:
+                schema: Error
+        """
+        data = self.get_json()
+        payload = data['xml']
+
+        schema = f'{os.path.dirname(__file__)}/../../utils/schema/VOEvent-v2.0.xsd'
+        voevent_schema = xmlschema.XMLSchema(schema)
+        if voevent_schema.is_valid(payload):
+            root = lxml.etree.fromstring(payload.encode('ascii'))
+        else:
+            raise Exception("xml file is not valid VOEvent")
+
+        dateobs = get_dateobs(root)
+
+        try:
+            event = GcnEvent.query.filter_by(dateobs=dateobs).one()
+        except NoResultFound:
+            event = DBSession().merge(GcnEvent(dateobs=dateobs))
+
+        tags = [GcnTag(dateobs=event.dateobs, text=_) for _ in get_tags(root)]
+
+        gcn_notice = GcnNotice(
+            content=payload.encode('ascii'),
+            ivorn=root.attrib['ivorn'],
+            notice_type=gcn.get_notice_type(root),
+            stream=urlparse(root.attrib['ivorn']).path.lstrip('/'),
+            date=root.find('./Who/Date').text,
+            dateobs=event.dateobs,
+        )
+
+        for tag in tags:
+            DBSession().merge(tag)
+        DBSession().merge(gcn_notice)
+
+        skymap = get_skymap(root, gcn_notice)
+        skymap["dateobs"] = event.dateobs
+
+        localization = Localization.query.filter_by(
+            dateobs=dateobs, localization_name=skymap["localization_name"]
+        ).all()
+        if len(localization) == 0:
+            DBSession().merge(Localization(**skymap))
+            DBSession().commit()
+        localization = Localization.query.filter_by(
+            dateobs=dateobs, localization_name=skymap["localization_name"]
+        ).one()
+
+        localization = get_contour(localization)
+        DBSession().merge(localization)
+        DBSession().commit()
+
+        return self.success()
+
+
+default_prefs = {'maxNumGcnEvents': 10, 'sinceDaysAgo': 3650}
+
+
+class GcnEventViewsHandler(BaseHandler):
+    @auth_or_token
+    def get(self):
+        user_prefs = getattr(self.current_user, 'preferences', None) or {}
+        top_events_prefs = user_prefs.get('topGcnEvents', {})
+        top_events_prefs = {**default_prefs, **top_events_prefs}
+
+        max_num_events = int(top_events_prefs['maxNumGcnEvents'])
+        since_days_ago = int(top_events_prefs['sinceDaysAgo'])
+
+        cutoff_day = datetime.datetime.now() - datetime.timedelta(days=since_days_ago)
+        q = GcnEvent.query
+
+        events = []
+        for event in q.all():
+            if len(events) >= max_num_events:
+                continue
+            dateobs = event.dateobs
+            if dateobs < cutoff_day:
+                continue
+            tags = [_ for _ in event.tags]
```

Comment:
```suggestion
            tags = [tag for tag in event.tags]
```
(`_` usually indicates that the variable goes unused)

Reply:
also, why not just `tags = event.tags`?